src/electronic-id.cpp

```diff
@@ -125,6 +137,36 @@ inline std::string byteVectorToHexString(const byte_vector& bytes)
     return hexStringBuilder.str();
 }
 
+bool matchATRWithMask(const byte_vector& atr, const byte_vector& pattern, const byte_vector& mask)
+{
+    if (pattern.size() != mask.size()) {
```

Comment:
I think here we can use std::array and static_assert to make it compile time error.

Reply:
Good point, thanks!

Reply:
As discussed, there is unfortunately no good way to implement this. Great idea nevertheless!

Reply:
But it still should be moved to MaskedATREntry constructor not executed in every pattern matching.